std/math.d

```diff
@@ -526,50 +555,48 @@ enum real SQRT1_2 =    SQRT2/2;                               /** $(SQRT)$(HALF)
  * For complex numbers, abs(z) = sqrt( $(POWER z.re, 2) + $(POWER z.im, 2) )
  * = hypot(z.re, z.im).
  */
-Num abs(Num)(Num x) @safe pure nothrow
-if ((is(typeof(Num.init >= 0)) && is(typeof(-Num.init)) ||
-    (is(Num == short) || is(Num == byte))) &&
-    !(is(Num* : const(ifloat*)) || is(Num* : const(idouble*))
-    || is(Num* : const(ireal*))))
+auto abs(Num)(Num x) @safe pure nothrow @nogc
+if (is(typeof(Num.init >= 0)) && is(typeof(-Num.init)) ||
+    (is(Num == short) || is(Num == byte)) || isOldComplex!Num || isComplex!Num)
 {
-    static if (isFloatingPoint!(Num))
+    static if (isComplex!Num)
+    {
+        return hypot(x.re, x.im);
+    }
+    else static if (isOldImaginary!Num)
+    {
+        return hypot(x.re, x.im);
+    }
+    else static if (isOldComplex!Num)
+    {
+        return fabs(x.im);
+    }
+    else static if (isFloatingPoint!Num)
+    {
         return fabs(x);
+    }
+    else static if (is(Num == short) || is(Num == byte))
+    {
+        return x >= 0 ? x : cast(Num) -int(x);
+    }
     else
     {
-        static if (is(Num == short) || is(Num == byte))
-            return x >= 0 ? x : cast(Num) -int(x);
-        else
-            return x >= 0 ? x : -x;
+        return x >= 0 ? x : -x;
     }
 }
 
-/// ditto
-auto abs(Num)(Num z) @safe pure nothrow @nogc
-if (is(Num* : const(cfloat*)) || is(Num* : const(cdouble*))
-    || is(Num* : const(creal*)))
-{
-    return hypot(z.re, z.im);
-}
-
-/// ditto
-auto abs(Num)(Num y) @safe pure nothrow @nogc
-if (is(Num* : const(ifloat*)) || is(Num* : const(idouble*))
-    || is(Num* : const(ireal*)))
-{
-    return fabs(y.im);
-}
-
 /// ditto
 @safe pure nothrow @nogc unittest
 {
+    import std.complex : complex;
     assert(isIdentical(abs(-0.0L), 0.0L));
     assert(isNaN(abs(real.nan)));
     assert(abs(-real.infinity) == real.infinity);
-    assert(abs(-3.2Li) == 3.2L);
```

Comment:
The complex deprecation currently triggers error in deprecated code. See https://issues.dlang.org/show_bug.cgi?id=18212